### PR TITLE
VS-530 VDS creation documentation for AoU

### PR DESCRIPTION
**Pull request by RoriCremer**
We now want to fully update the AoU documentation so that a VDS can be created  (no VCFs!)

### CODE REVIEW

File: scripts/variantstore/docs/aou/AOU_DELIVERABLES.md [hunk 1]
```diff
@@ -13,12 +13,10 @@
   - [GvsWithdrawSamples](https://dockstore.org/my-workflows/github.com/broadinstitute/gatk/GvsWithdrawSamples) workflow
   - [GvsPopulateAltAllele](https://dockstore.org/my-workflows/github.com/broadinstitute/gatk/GvsPopulateAltAllele) workflow
   - [GvsCreateFilterSet](https://dockstore.org/my-workflows/github.com/broadinstitute/gatk/GvsCreateFilterSet) workflow
-  - [GvsPrepareRangesCallset](https://dockstore.org/my-workflows/github.com/broadinstitute/gatk/GvsPrepareRangesCallset) workflow (VCF output)
-  - [GvsExtractCallset](https://dockstore.org/my-workflows/github.com/broadinstitute/gatk/GvsExtractCallset) workflow (VCF output)
+  - [GvsExtractAvroFilesForHail](https://dockstore.org/my-workflows/github.com/broadinstitute/gatk/GvsExtractAvroFilesForHail) workflow (Hail VDS output)
```
Comment (rsasch) [inline scripts/variantstore/docs/aou/AOU_DELIVERABLES.md:16]: no need to mention VDS since this doc is now only VDS output
```suggestion
  - [GvsExtractAvroFilesForHail](https://dockstore.org/my-workflows/github.com/broadinstitute/gatk/GvsExtractAvroFilesForHail) workflow
```

File: scripts/variantstore/docs/aou/AOU_DELIVERABLES.md [hunk 2]
```diff
@@ -13,12 +13,10 @@
...
+  - [GvsPrepareRangesCallset](https://dockstore.org/my-workflows/github.com/broadinstitute/gatk/GvsPrepareRangesCallset) workflow (Hail VDS output -- used for callset stats)
```
Comment (rsasch) [inline scripts/variantstore/docs/aou/AOU_DELIVERABLES.md:17]: see above
```suggestion
  - [GvsPrepareRangesCallset](https://dockstore.org/my-workflows/github.com/broadinstitute/gatk/GvsPrepareRangesCallset) workflow
```

File: scripts/variantstore/docs/aou/AOU_DELIVERABLES.md [hunk 3]
```diff
@@ -52,26 +50,32 @@
...
    - See [naming conventions doc](https://docs.google.com/document/d/1pNtuv7uDoiOFPbwe4zx5sAGH7MyxwKqXkyrpNmBxeow) for guidance on what to use for `filter_set_name`, which you will need to keep track of for the `GvsExtractCallset` WDL. If, for some reason, this step needs to be run multiple times, be sure to use a different `filter_set_name` (the doc has guidance for this, as well).
    - This workflow does not use the Terra Data Entity Model to run, so be sure to select the `Run workflow with inputs defined by file paths` workflow submission option.
-7. `GvsPrepareRangesCallset` workflow
-   - This workflow transforms the data in the vet and ref_ranges tables into a schema optimized for VCF generation during the Extract step.
-   - It will need to be run twice, once with `control_samples` set to "true" (see [naming conventions doc](https://docs.google.com/document/d/1pNtuv7uDoiOFPbwe4zx5sAGH7MyxwKqXkyrpNmBxeow) for guidance on what to use for `extract_table_prefix` or cohort prefix, which you will need to keep track of for the `GvsExtractCallset` WDL); the default value is `false`.
+7. `GvsExtractAvroFilesForHail` workflow (Hail VDS output)
```
Comment (rsasch) [inline scripts/variantstore/docs/aou/AOU_DELIVERABLES.md:53]: see above
```suggestion
7. `GvsExtractAvroFilesForHail` workflow    
```

File: scripts/variantstore/docs/aou/AOU_DELIVERABLES.md [hunk 4]
```diff
@@ -52,26 +50,32 @@
...
    - This workflow does not use the Terra Data Entity Model to run, so be sure to select the `Run workflow with inputs defined by file paths` workflow submission option.
-7. `GvsPrepareRangesCallset` workflow
-   - This workflow transforms the data in the vet and ref_ranges tables into a schema optimized for VCF generation during the Extract step.
-   - It will need to be run twice, once with `control_samples` set to "true" (see [naming conventions doc](https://docs.google.com/document/d/1pNtuv7uDoiOFPbwe4zx5sAGH7MyxwKqXkyrpNmBxeow) for guidance on what to use for `extract_table_prefix` or cohort prefix, which you will need to keep track of for the `GvsExtractCallset` WDL); the default value is `false`.
+7. `GvsExtractAvroFilesForHail` workflow (Hail VDS output)    
+   - This workflow extracts the data in BigQuery and transforms it into Avro files in a Google bucket, incorporating the VQSR filter set data.  We suggest this method for callsets of more than 100k samples.
```
Comment (rsasch) [inline scripts/variantstore/docs/aou/AOU_DELIVERABLES.md:54]: basically, the same as above
```suggestion
   - This workflow extracts the data in BigQuery and transforms it into Avro files in a Google bucket, incorporating the VQSR filter set data.
```

File: scripts/variantstore/docs/aou/AOU_DELIVERABLES.md [hunk 5]
```diff
@@ -52,26 +50,32 @@
...
+   - The extracted Avro files will then be used as the input for the below notebook which can then be run to produce a Hail VDS
```
Comment (rsasch) [inline scripts/variantstore/docs/aou/AOU_DELIVERABLES.md:55]: nits
```suggestion
   - The extracted Avro files will then be used as the inputs for the below notebook which will then be used to produce a Hail VDS.
```

File: scripts/variantstore/docs/aou/AOU_DELIVERABLES.md [hunk 6]
```diff
@@ -52,26 +50,32 @@
...
    - This workflow does not use the Terra Data Entity Model to run, so be sure to select the `Run workflow with inputs defined by file paths` workflow submission option.
-8. `GvsExtractCallset` workflow
-   - This workflow extracts the data in BigQuery and transforms it into a sharded joint called VCF incorporating the VQSR filter set data.  We will probably not run this on callsets of more than 100K samples.
-   - It also needs to be run twice, once with `control_samples` set to "true", and with the `filter_set_name` and `extract_table_prefix` from step 5 & 6.  Include a valid (and secure) "output_gcs_dir" parameter, which is where the VCF, interval list, manifest, and sample name list files will go.
+8. Run the VDS Extract using a notebook terminal and the python script `hail_gvs_import.py`
```
Comment (rsasch) [inline scripts/variantstore/docs/aou/AOU_DELIVERABLES.md:57]: Where is this script file located?

File: scripts/variantstore/docs/aou/AOU_DELIVERABLES.md [hunk 7]
```diff
@@ -52,26 +50,32 @@
...
+   - This step creates a VDS based on the Avro files 
+   - Notebook provisioning suggestions can be found in the file: [AoU Delta VDS Cluster Configuration.md](vds/cluster/AoU Delta VDS Cluster Configuration.md)
```
Comment (rsasch) [inline scripts/variantstore/docs/aou/AOU_DELIVERABLES.md:59]: link did not work when I viewed the file in Github
```suggestion
   - Notebook provisioning suggestions can be found in the file: [AoU Delta VDS Cluster Configuration.md](vds/cluster/AoU%20Delta%20VDS%20Cluster%20Configuration.md)
```

File: scripts/variantstore/docs/aou/AOU_DELIVERABLES.md [hunk 8]
```diff
@@ -52,26 +50,32 @@
...
+   - `hail_gvs_import.py` is the python script that needs to be run in the notebook-- we suggest running it directly in the terminal
```
Comment (rsasch) [inline scripts/variantstore/docs/aou/AOU_DELIVERABLES.md:60]: this seems redundant

File: scripts/variantstore/docs/aou/AOU_DELIVERABLES.md [hunk 9]
```diff
@@ -52,26 +50,32 @@
...
+   - inputs to the python script are:
+     1. `--avro-path`: the directory path at which exported GVS Avro files are found in GCP
+     2. `--vds-path`: the desired output path to which the VDS should be written
+     3. `--temp-path`: a convenient path to temporary directory (we suggest the GCP bucket of the workspace that the notebook is in)
```
Comment (rsasch) [inline scripts/variantstore/docs/aou/AOU_DELIVERABLES.md:64]: Do these path require (or break with) a leading "/"?

File: scripts/variantstore/docs/aou/AOU_DELIVERABLES.md [hunk 10]
```diff
@@ -52,26 +50,32 @@
...
-   - It also needs to be run twice, once with `control_samples` set to "true", and with the `filter_set_name` and `extract_table_prefix` from step 5 & 6.  Include a valid (and secure) "output_gcs_dir" parameter, which is where the VCF, interval list, manifest, and sample name list files will go.
+8. Run the VDS Extract using a notebook terminal and the python script `hail_gvs_import.py`
+   - This step creates a VDS based on the Avro files 
+   - Notebook provisioning suggestions can be found in the file: [AoU Delta VDS Cluster Configuration.md](vds/cluster/AoU Delta VDS Cluster Configuration.md)
+   - `hail_gvs_import.py` is the python script that needs to be run in the notebook-- we suggest running it directly in the terminal
+   - inputs to the python script are:
+     1. `--avro-path`: the directory path at which exported GVS Avro files are found in GCP
+     2. `--vds-path`: the desired output path to which the VDS should be written
+     3. `--temp-path`: a convenient path to temporary directory (we suggest the GCP bucket of the workspace that the notebook is in) 
+9. `GvsPrepareRangesCallset` workflow (Hail VDS output -- for callset stats)
```
Comment (rsasch) [inline scripts/variantstore/docs/aou/AOU_DELIVERABLES.md:65]: see above
```suggestion
9. `GvsPrepareRangesCallset` workflow
```

File: scripts/variantstore/docs/aou/AOU_DELIVERABLES.md [hunk 11]
```diff
@@ -52,26 +50,32 @@
...
+   - This workflow transforms the data in the vet tables into a schema optimized for VCF generation during the Extract step.
+   - It will need to be run once with `only_output_vet_tables` set to "true" (see [naming conventions doc](https://docs.google.com/document/d/1pNtuv7uDoiOFPbwe4zx5sAGH7MyxwKqXkyrpNmBxeow) for guidance on what to use for `extract_table_prefix` or cohort prefix, which you will need to keep track of for the callset stats); the default value is `false`.
```
Comment (rsasch) [inline scripts/variantstore/docs/aou/AOU_DELIVERABLES.md:67]:
```suggestion
   - It will need to be run after the VDS has been successfully created with `only_output_vet_tables` set to "true" (see [naming conventions doc](https://docs.google.com/document/d/1pNtuv7uDoiOFPbwe4zx5sAGH7MyxwKqXkyrpNmBxeow) for guidance on what to use for `extract_table_prefix` or cohort prefix); the default value is `false`.
```

File: scripts/variantstore/docs/aou/AOU_DELIVERABLES.md [hunk 12]
```diff
@@ -52,26 +50,32 @@
...
-     - "BigQuery Data Viewer" role for your @pmi-ops proxy group on the `spec-ops-aou:gvs_public_reference_data.gnomad_v3_sites` table
-     - the Google project ID you used for all the GVS WDLs (`project_id` input)
-     - the name of the BigQuery dataset you used for all the GVS WDLs (`dataset_name` input)
-     - the `extract_table_prefix` input from `GvsExtractCallset` step
-     - the `filter_set_name` input from `GvsCreateFilterSet` step
-12. [GvsCalculatePrecisionAndSensitivity](../../tieout/AoU_PRECISION_SENSITIVITY.md#generating-callset-precision-and-sensitivity-values) workflow, for which you need
-    - "Storage Object View" access granted for your @pmi-ops proxy group on the `gs://broad-dsp-spec-ops/gvs/truth` directory
-13. `GvsCallsetCost`
-    - The cost from this callset, which represents the total BigQuery cost (which is not represented in the Terra UI total workflow cost) from the GVS pipeline workflows, is used to calculate the cost of the callset as a whole and by sample.
+10. Run the notebook to create callset stats (**TODO**, see [VS-388](https://broadworkbench.atlassian.net/browse/VS-388)), for which you need
```
Comment (rsasch) [inline scripts/variantstore/docs/aou/AOU_DELIVERABLES.md:69]:
```suggestion
10. `GvsCallsetStatistics` workflow
```

File: scripts/variantstore/docs/aou/AOU_DELIVERABLES.md [hunk 13]
```diff
@@ -52,26 +50,32 @@
...
+   - "BigQuery Data Viewer" role for your @pmi-ops proxy group on the `spec-ops-aou:gvs_public_reference_data.gnomad_v3_sites` table
+   - the Google project ID you used for all the GVS WDLs (`project_id` input)
+   - the name of the BigQuery dataset you used for all the GVS WDLs (`dataset_name` input)
+   - the `extract_table_prefix` input from `GvsExtractCallset` step
+   - the `filter_set_name` input from `GvsCreateFilterSet` step
```
Comment (rsasch) [inline scripts/variantstore/docs/aou/AOU_DELIVERABLES.md:74]: nit: the formatting/structure for these should be the same as the other workflows

File: scripts/variantstore/docs/aou/AOU_DELIVERABLES.md [hunk 14]
```diff
@@ -52,26 +50,32 @@
...
+11. [GvsCalculatePrecisionAndSensitivity](../../tieout/AoU_PRECISION_SENSITIVITY.md#generating-callset-precision-and-sensitivity-values) workflow, for which you need
```
Comment (rsasch) [inline scripts/variantstore/docs/aou/AOU_DELIVERABLES.md:75]:
```suggestion
11. `GvsCalculatePrecisionAndSensitivity.wdl` workflow
```

File: scripts/variantstore/docs/aou/AOU_DELIVERABLES.md [hunk 15]
```diff
@@ -52,26 +50,32 @@
...
+   - "Storage Object View" access granted for your @pmi-ops proxy group on the `gs://broad-dsp-spec-ops/gvs/truth` directory
```
Comment (rsasch) [inline scripts/variantstore/docs/aou/AOU_DELIVERABLES.md:76]: nit: to make it consistent with format/structure of other steps
```suggestion
    - you will need to have "Storage Object View" access granted for your @pmi-ops proxy group on the `gs://broad-dsp-spec-ops/gvs/truth` directory
```

File: scripts/variantstore/docs/aou/AOU_DELIVERABLES.md [hunk 16]
```diff
@@ -52,26 +50,32 @@
...
+12. `GvsCallsetCost`
```
Comment (rsasch) [inline scripts/variantstore/docs/aou/AOU_DELIVERABLES.md:77]: see above
```suggestion
12. `GvsCallsetCost.wdl` workflow
```

File: scripts/variantstore/docs/aou/AOU_DELIVERABLES.md [hunk 17]
```diff
@@ -52,26 +50,32 @@
...
+   - The cost from this callset, which represents the total BigQuery cost (which is not represented in the Terra UI total workflow cost) from the GVS pipeline workflows, is used to calculate the cost of the callset as a whole and by sample.
```
Comment (rsasch) [inline scripts/variantstore/docs/aou/AOU_DELIVERABLES.md:78]:
```suggestion
   - This workflow calculates the total BigQuery cost of generating this callset (which is not represented in the Terra UI total workflow cost) using the above GVS workflows; it's used to calculate the cost as a whole and by sample.
```

File: scripts/variantstore/wdl/extract/hail_gvs_import.py [hunk 18]
```diff
@@ -30,7 +30,7 @@ def import_gvs(argsfn, vds_path, references_path, temp_path):
     rg38.add_sequence(f'{references_path}/Homo_sapiens_assembly38.fasta.gz',
                       f'{references_path}/Homo_sapiens_assembly38.fasta.fai')
 
-
+    ## note that a full description of the import_gvs function written by Hail for this process can be found here: hail/python/hail/methods/impex.py
```
Comment (rsasch) [inline scripts/variantstore/wdl/extract/hail_gvs_import.py:33]: what is the full path to the file?

File: scripts/variantstore/wdl/extract/hail_gvs_import.py [hunk 19]
```diff
@@ -41,6 +41,12 @@ def import_gvs(argsfn, vds_path, references_path, temp_path):
         reference_genome=rg38,
         final_path=vds_path,
         tmp_dir=f'{temp_path}/hail_tmp_import_gvs',
+        # truth_sensitivity_snp_threshold: 'float' = 0.997,
+        # truth_sensitivity_indel_threshold: 'float' = 0.990,
+        # partitions_per_sample=0.35,
+        # intermediate_resume_point=0,
+        # skip_final_merge=false,
+        # unphase=false,
```
Comment (rsasch) [inline scripts/variantstore/wdl/extract/hail_gvs_import.py:49]: Why is this here?

**Review by rsasch — CHANGES_REQUESTED** [review]
Some small edits for clarity...etc. before actually trying to generate the VDS.

File: scripts/variantstore/wdl/extract/hail_gvs_import.py (hunk 19 above)
Comment (RoriCremer) [reply scripts/variantstore/wdl/extract/hail_gvs_import.py:49]: so that if you want to adjust this python script a bit for different defaults (eg you dont want phasing info) you can see this comment instead of looking in Tim's source code

File: scripts/variantstore/wdl/extract/hail_gvs_import.py (hunk 18 above)
Comment (RoriCremer) [reply scripts/variantstore/wdl/extract/hail_gvs_import.py:33]: this is in an entirely different repo so I'm actually not sure how to reference it at all
it's here: https://github.com/hail-is/hail/pull/12017/files#

Comment (rsasch) [reply scripts/variantstore/wdl/extract/hail_gvs_import.py:33]: if this file only exists as part of a branch, you can still reference it; in this case, the link would be https://github.com/tpoterba/hail/blob/import-gvs/hail/python/hail/methods/impex.py

File: scripts/variantstore/docs/aou/AOU_DELIVERABLES.md (hunk 9 above)
Comment (rsasch) [reply scripts/variantstore/docs/aou/AOU_DELIVERABLES.md:64]: All three paths correspond to directories; do you know if they are expecting a leading "/" (e.g. "gs://my-bucket/my-dir/") or not (e.g. "gs://my-bucket/my-dir")?

File: scripts/variantstore/docs/aou/AOU_DELIVERABLES.md [hunk 20]
```diff
@@ -52,27 +50,32 @@
...
    - This workflow does not use the Terra Data Entity Model to run, so be sure to select the `Run workflow with inputs defined by file paths` workflow submission option.
-8. `GvsExtractCallset` workflow
-   - This workflow extracts the data in BigQuery and transforms it into a sharded joint called VCF incorporating the VQSR filter set data.  We will probably not run this on callsets of more than 100K samples.
-   - It also needs to be run twice, once with `control_samples` set to "true", and with the `filter_set_name` and `extract_table_prefix` from step 5 & 6.  Include a valid (and secure) "output_gcs_dir" parameter, which is where the VCF, interval list, manifest, and sample name list files will go.
+8. Run the VDS Extract using a notebook terminal and the python script [hail_gvs_import.py](scripts/variantstore/wdl/extract/hail_gvs_import.py)
+   - This step creates a VDS based on the Avro files 
+   - Notebook provisioning suggestions can be found in the file: [AoU Delta VDS Cluster Configuration.md](vds/cluster/AoU%20Delta%20VDS%20Cluster%20Configuration.md)
+   - We suggest running `hail_gvs_import.py` directly in the terminal
```
Comment (rsasch) [inline scripts/variantstore/docs/aou/AOU_DELIVERABLES.md:60]: nit
```suggestion
   - We suggest gsutil cp-ing `hail_gvs_import.py` to the notebook and then invoke it directly in the terminal
```

File: scripts/variantstore/docs/aou/AOU_DELIVERABLES.md [hunk 21]
```diff
@@ -52,27 +50,32 @@
...
    - This workflow does not use the Terra Data Entity Model to run, so be sure to select the `Run workflow with inputs defined by file paths` workflow submission option.
-8. `GvsExtractCallset` workflow
-   - This workflow extracts the data in BigQuery and transforms it into a sharded joint called VCF incorporating the VQSR filter set data.  We will probably not run this on callsets of more than 100K samples.
-   - It also needs to be run twice, once with `control_samples` set to "true", and with the `filter_set_name` and `extract_table_prefix` from step 5 & 6.  Include a valid (and secure) "output_gcs_dir" parameter, which is where the VCF, interval list, manifest, and sample name list files will go.
+8. Run the VDS Extract using a notebook terminal and the python script [hail_gvs_import.py](scripts/variantstore/wdl/extract/hail_gvs_import.py)
+   - This step creates a VDS based on the Avro files 
+   - Notebook provisioning suggestions can be found in the file: [AoU Delta VDS Cluster Configuration.md](vds/cluster/AoU%20Delta%20VDS%20Cluster%20Configuration.md)
```
Comment (rsasch) [inline scripts/variantstore/docs/aou/AOU_DELIVERABLES.md:59]:
```suggestion
   - Notebook provisioning suggestions and other necessary set up details (e.g. the Hail wheel to use) can be found in the file: [AoU Delta VDS Cluster Configuration.md](vds/cluster/AoU%20Delta%20VDS%20Cluster%20Configuration.md)
```

File: scripts/variantstore/docs/aou/AOU_DELIVERABLES.md (hunk 9 above)
Comment (mcovarr) [reply scripts/variantstore/docs/aou/AOU_DELIVERABLES.md:64]: They're all okay [with or without slashes](https://github.com/broadinstitute/gatk/blob/aa97a09a4467a1ff38892d4af6aa34463bf3ddac/scripts/variantstore/wdl/extract/hail_gvs_import.py#L134-L136).

File: scripts/variantstore/wdl/extract/hail_gvs_import.py [hunk 22]
```diff
@@ -30,7 +30,8 @@ def import_gvs(argsfn, vds_path, references_path, temp_path):
     rg38.add_sequence(f'{references_path}/Homo_sapiens_assembly38.fasta.gz',
                       f'{references_path}/Homo_sapiens_assembly38.fasta.fai')
 
-
+    ## Note that a full description of the import_gvs function written by Hail for this process can be found here: hail/python/hail/methods/impex.py
```
Comment (mcovarr) [inline scripts/variantstore/wdl/extract/hail_gvs_import.py:33]: Here's a [reference to the file](https://github.com/tpoterba/hail/blob/import-gvs/hail/python/hail/methods/impex.py#L3022) as it currently exists only in Tim's fork and not in the main Hail repo.

File: scripts/variantstore/docs/aou/AOU_DELIVERABLES.md (hunk 9 above)
Comment (mcovarr) [reply scripts/variantstore/docs/aou/AOU_DELIVERABLES.md:64]: As for `--temp-path` I would probably *not* suggest the bare bucket path as that might make a mess:

```suggestion
     1. `--avro-path`: the directory path at which exported GVS Avro files are found in GCP
     2. `--vds-path`: the desired output path to which the VDS should be written
     3. `--temp-path`: a convenient path to temporary directory. We suggest a folder under the GCP bucket of the workspace that the notebook is in, e.g. `gs://fc-<workspace-id>/hail_tmp`.
```

File: scripts/variantstore/wdl/extract/hail_gvs_import.py (hunk 22 above)
Comment (RoriCremer) [reply scripts/variantstore/wdl/extract/hail_gvs_import.py:33]: so I should remove this? I dont understand...

File: scripts/variantstore/docs/aou/AOU_DELIVERABLES.md [hunk 23]
```diff
@@ -52,27 +50,32 @@
...
-13. `GvsCallsetCost`
-    - The cost from this callset, which represents the total BigQuery cost (which is not represented in the Terra UI total workflow cost) from the GVS pipeline workflows, is used to calculate the cost of the callset as a whole and by sample.
-
+8. Run the VDS Extract using a notebook terminal and the python script called `hail_gvs_import.py` from the GenerateHailScripts task. It will look something like: `gs://fc-<workspace-id>/submissions/<submission id>/GvsExtractAvroFilesForHail/<workflow id>/call-GenerateHailScripts/hail_gvs_import.py`
+    - This step creates a VDS based on the Avro files 
+    - Notebook provisioning suggestions and other necessary set up details (e.g. the Hail wheel to use) can be found in the file: [AoU Delta VDS Cluster Configuration.md](vds/cluster/AoU%20Delta%20VDS%20Cluster%20Configuration.md)
+    - We suggest gsutil cp-ing `hail_gvs_import.py` to the notebook and then invoking it directly in the terminal
+    - inputs to the python script are:
+      1. `--avro-path`: the directory path at which exported GVS Avro files are found in GCP
```
Comment (mcovarr) [inline scripts/variantstore/docs/aou/AOU_DELIVERABLES.md:62]: the version of the script generated at `GenerateHailScripts/hail_gvs_import.py` should have set these to the correct defaults

Comment (RoriCremer) [reply scripts/variantstore/docs/aou/AOU_DELIVERABLES.md:63]: this needs to be investigated---it doesn't seem to be the case currently?

**Review by rsasch — CHANGES_REQUESTED** [review]

File: scripts/variantstore/docs/aou/AOU_DELIVERABLES.md [hunk 24]
```diff
@@ -50,41 +43,49 @@
...
+    - This workflow calculates the total BigQuery cost of generating this callset (which is not represented in the Terra UI total workflow cost) using the above GVS workflows; it's used to calculate the cost as a whole and by sample.
+    - Note that the VDS creation cost based on the VM time will have to be added and the Avro extraction was not included at the time of this readme note
 
 ## Deliverables (via email to stakeholders once the above steps are complete)
-1. GCS locations of the VCFs, indexes and interval_list files (subpaths of the `output_gcs_dir` input from `GvsExtractCallset`)
+1. GCS location of the VDS (the `--vds-path` input from `hail_gvs_import.py`)
 2. fully qualified name of the BigQuery dataset (composed of the `project_id` and `dataset_name` inputs from the workflows)
-3. callset statistics CSV file (see step #11 above)
+3. CSV output from `GvsCallsetStatistics` workflow (see step #11 above)
 4. TSV output from `GvsCalculatePrecisionAndSensitivity` workflow (see step #12 above)
```
[the remark this answers is not in the document]
Comment (RoriCremer) [reply scripts/variantstore/docs/aou/AOU_DELIVERABLES.md:82]: hahaha well this has been wrong for a while then, or I deleted the real step 12?

[the remark this answers is not in the document]
Comment (RoriCremer) [reply scripts/variantstore/docs/aou/AOU_DELIVERABLES.md:82]: I will add it back lol

File: scripts/variantstore/docs/aou/AOU_DELIVERABLES.md [hunk 25]
```diff
@@ -52,27 +50,32 @@
...
-8. `GvsExtractCallset` workflow
-   - This workflow extracts the data in BigQuery and transforms it into a sharded joint called VCF incorporating the VQSR filter set data.  We will probably not run this on callsets of more than 100K samples.
-   - It also needs to be run twice, once with `control_samples` set to "true", and with the `filter_set_name` and `extract_table_prefix` from step 5 & 6.  Include a valid (and secure) "output_gcs_dir" parameter, which is where the VCF, interval list, manifest, and sample name list files will go.
+8. Run the VDS Extract using a notebook terminal and the python script [hail_gvs_import.py](scripts/variantstore/wdl/extract/hail_gvs_import.py)
+   - This step creates a VDS based on the Avro files 
+   - Notebook provisioning suggestions can be found in the file: [AoU Delta VDS Cluster Configuration.md](vds/cluster/AoU%20Delta%20VDS%20Cluster%20Configuration.md)
+   - We suggest running `hail_gvs_import.py` directly in the terminal
+   - inputs to the python script are:
+     1. `--avro-path`: the directory path at which exported GVS Avro files are found in GCP
```
[the remark this answers is not in the document]
Comment (mcovarr) [reply scripts/variantstore/docs/aou/AOU_DELIVERABLES.md:59]: Per our earlier discussion the script generation here is definitely broken; the replacement logic is looking for the wrong values in the input template. Created [VS-809](https://broadworkbench.atlassian.net/browse/VS-809) to fix this.

File: scripts/variantstore/docs/aou/AOU_DELIVERABLES.md [hunk 26]
```diff
@@ -50,41 +45,55 @@
...
+12. `GvsCallsetCost` workflow
+    - This workflow calculates the total BigQuery cost of generating this callset (which is not represented in the Terra UI total workflow cost) using the above GVS workflows; it's used to calculate the cost as a whole and by sample.
 
 ## Deliverables (via email to stakeholders once the above steps are complete)
-1. GCS locations of the VCFs, indexes and interval_list files (subpaths of the `output_gcs_dir` input from `GvsExtractCallset`)
+1. GCS location of the VDS (the `--vds-path` input from `hail_gvs_import.py`)
 2. fully qualified name of the BigQuery dataset (composed of the `project_id` and `dataset_name` inputs from the workflows)
-3. callset statistics CSV file (see step #11 above)
-4. TSV output from `GvsCalculatePrecisionAndSensitivity` workflow (see step #12 above)
+3. CSV output from `GvsCallsetStatistics` workflow
```
Comment (rsasch) [inline scripts/variantstore/docs/aou/AOU_DELIVERABLES.md:89]: Probably a good idea to link to vds/delivery/Delivering%20a%20VDS.md since that includes details of how to get the VDS to the AoU delivery bucket, naming conventions...etc.

Comment (RoriCremer) [reply scripts/variantstore/docs/aou/AOU_DELIVERABLES.md]: great idea

File: scripts/variantstore/docs/aou/AOU_DELIVERABLES.md [hunk 27]
```diff
@@ -50,41 +45,61 @@
...
-3. callset statistics CSV file (see step #11 above)
-4. TSV output from `GvsCalculatePrecisionAndSensitivity` workflow (see step #12 above)
+1. GCS location of the VDS (the `--vds-path` input from `hail_gvs_import.py`)
+2. Fully qualified name of the BigQuery dataset (composed of the `project_id` and `dataset_name` inputs from the workflows)
+3. CSV output from `GvsCallsetStatistics` workflow
+4. TSV output from `GvsCalculatePrecisionAndSensitivity` workflow
+
+To get the VDS someplace secure and accessible for delivery, see [Delivering a VDS](vds/delivery/Delivering%20a%20VDS.md)
+This includes details of how to get the VDS to the AoU delivery bucket as well as VDS naming conventions.
+The Callset Stats and S&P files can be simply gsutil cp-ed to the AoU delivery bucket since they are so much smaller.
```
Comment (rsasch) [inline scripts/variantstore/docs/aou/AOU_DELIVERABLES.md:94]: nit
```suggestion
To get the VDS someplace secure and accessible for delivery, see [Delivering a VDS](vds/delivery/Delivering%20a%20VDS.md)
This includes details of how to get the VDS to the AoU delivery bucket as well as VDS naming conventions.
The Callset Stats and S&P files can be simply gsutil cp-ed to the AoU delivery bucket since they are so much smaller.
1. GCS location of the VDS in the AoU delivery bucket
2. Fully qualified name of the BigQuery dataset (composed of the `project_id` and `dataset_name` inputs from the workflows)
3. GCS location of the CSV output from `GvsCallsetStatistics` workflow in the AoU delivery bucket
4. GCS location of the TSV output from `GvsCalculatePrecisionAndSensitivity` in the AoU delivery bucket
```

**Review by rsasch — APPROVED** [review]
One nit to re-arrange it a bit, otherwise 👍🏻
